packages/spl/src/layout-utils.ts

```diff
@@ -57,7 +57,7 @@ export class BorshString extends Layout<string> {
 
   getSpan(b: Uint8Array, offset = 0): number {
     if (!b) {
-      return this.maxLength
+      return u32().span + this.maxLength
```

Comment:
This feels like a magic fix. Can you add a comment or at least comment here on what this is fixing?

Reply:
it's actually more just semantics - previously, specifying a "max size" for the "BorshString" class was the max _encoded_ size, meaning that it capped the size of the resulting encoded value. With this change, the "max size" becomes the max length of the actual _string_ value.

This aligns more closely to what was used (and why that bug was introduced) since using max length was meant to ensure that the _string_ value was less than 32 bytes for the sake of generating account address seeds from it.